admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-choices-selector.ts

```diff
@@ -59,21 +87,38 @@ export default class BulkChoicesSelector {
    * Delegated event listener on tabContainer, because every checkbox is re-rendered with dynamic pagination
    */
   private listenCheckboxesChange(): void {
+    this.eventEmitter.on(CombinationEvents.listRendered, () => {
+      const bulkSelectedAll = this.tabContainer
+        .querySelector<HTMLInputElement>(`${CombinationMap.bulkSelectAllCheckboxes}:checked`);
+      this.checkAll(!!bulkSelectedAll);
+    });
+
     this.tabContainer.addEventListener('change', (e) => {
       const checkbox = e.target;
 
       if (!(checkbox instanceof HTMLInputElement)) {
         return;
       }
 
-      const isBulkSelectAll = checkbox.matches(CombinationMap.bulkSelectAll);
+      const isBulkSelectAll = checkbox.matches(`${CombinationMap.bulkSelectAllCheckboxes}`);
```

Comment:
Not sure the `` notation is worth it here since it only contains `CombinationMap.bulkSelectAllCheckboxes` anyway